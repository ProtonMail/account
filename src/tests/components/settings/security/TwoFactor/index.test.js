import TwoFactorSettings from '../../../../../components/settings/security/TwoFactor';
import { renderProvided } from '../../../../testsHelpers/storeTools';

describe('TwoFactorSettings', () => {
    test('2FA disabled', () => {
        expect(renderProvided(<TwoFactorSettings TwoFactor={0} TOTP={0}/>)).toMatchSnapshot();
    });

    test('TOTP only enable', () => {
        expect(renderProvided(<TwoFactorSettings TwoFactor={1} TOTP={1} U2FKeys={[]}/>)).toMatchSnapshot();
    });

    test('U2F only enable', () => {
        const keys = [
            { KeyHandle: '1', Compromised: false, Label: 'First key' },
            { KeyHandle: '2', Compromised: false, Label: 'Second key' },
            { KeyHandle: '3', Compromised: true, Label: 'Compromised key' },
            { KeyHandle: '4', Compromised: false, Label: 'Last key' }
        ];

        expect(renderProvided(<TwoFactorSettings TwoFactor={1} TOTP={0} U2FKeys={keys}/>)).toMatchSnapshot();
    });

    test('Everything enable', () => {
        const keys = [
            { KeyHandle: '1', Compromised: false, Label: 'First key' },
            { KeyHandle: '2', Compromised: false, Label: 'Second key' },
            { KeyHandle: '3', Compromised: true, Label: 'Compromised key' },
            { KeyHandle: '4', Compromised: false, Label: 'Last key' }
        ];

        expect(renderProvided(<TwoFactorSettings TwoFactor={1} TOTP={1} U2FKeys={keys}/>)).toMatchSnapshot();
    });
});
