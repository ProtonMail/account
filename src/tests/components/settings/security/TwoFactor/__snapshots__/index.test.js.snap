// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwoFactorSettings 2FA disabled 1`] = `
<div class="twoFactor">
  <h2>Two-Factor Authentication</h2>
  <div class="alert alert-info">Two-factor authentication is currently off.</div>
  <div id="totp" class="item">
    <div class="description">2FA via Application</div>
    <button class="action">Enable</button>
  </div>
  <div class="item">
    <div class="description">2FA via Security Key</div>
    <button class="action">Enable</button>
  </div>
  <ul class="list"></ul>
</div>
`;

exports[`TwoFactorSettings Everything enable 1`] = `
<div class="twoFactor">
  <h2>Two-Factor Authentication</h2>
  <div class="alert alert-info">Two-factor authentication is currently on.
    <button type="button" class="button">Turn off</button>
  </div>
  <div id="totp" class="item">
    <div class="description">2FA via Application</div>
    <button class="action">Disable</button>
    <div class="description">
      <button type="button" class="button">Regenerate recovery codes</button>
      <i data-toggle="tooltip" data-placement="top"
      title="Regenerate recovery codes will invalidate existing recovery codes">i</i>
    </div>
  </div>
  <div class="item lastItem">
    <div class="description">2FA via Security Key</div>
    <button class="action">Add another key</button>
  </div>
  <ul class="list">
    <li class="listElement">
      <div class="listElementHeader">First key</div>
      <button>Delete</button>
    </li>
    <li class="listElement">
      <div class="listElementHeader">Second key</div>
      <button>Delete</button>
    </li>
    <li class="listElement">
      <div class="listElementHeader listElementHeaderCompromised">Compromised key</div>
      <div class="badge badge-danger">Compromised</div>
      <button>Delete</button>
    </li>
    <li class="listElement">
      <div class="listElementHeader">Last key</div>
      <button>Delete</button>
    </li>
  </ul>
</div>
`;

exports[`TwoFactorSettings TOTP only enable 1`] = `
<div class="twoFactor">
  <h2>Two-Factor Authentication</h2>
  <div class="alert alert-info">Two-factor authentication is currently on.
    <button type="button" class="button">Turn off</button>
  </div>
  <div id="totp" class="item">
    <div class="description">2FA via Application</div>
    <button class="action">Disable</button>
    <div class="description">
      <button type="button" class="button">Regenerate recovery codes</button>
      <i data-toggle="tooltip" data-placement="top"
      title="Regenerate recovery codes will invalidate existing recovery codes">i</i>
    </div>
  </div>
  <div class="item lastItem">
    <div class="description">2FA via Security Key</div>
    <button class="action">Enable</button>
  </div>
  <ul class="list"></ul>
</div>
`;

exports[`TwoFactorSettings U2F only enable 1`] = `
<div class="twoFactor">
  <h2>Two-Factor Authentication</h2>
  <div class="alert alert-info">Two-factor authentication is currently on.
    <button type="button" class="button">Turn off</button>
  </div>
  <div id="totp" class="item">
    <div class="description">2FA via Application</div>
    <button class="action">Enable</button>
    <div class="description">
      <button type="button" class="button">Regenerate recovery codes</button>
      <i data-toggle="tooltip" data-placement="top"
      title="Regenerate recovery codes will invalidate existing recovery codes">i</i>
    </div>
  </div>
  <div class="item lastItem">
    <div class="description">2FA via Security Key</div>
    <button class="action">Add another key</button>
  </div>
  <ul class="list">
    <li class="listElement">
      <div class="listElementHeader">First key</div>
      <button>Delete</button>
    </li>
    <li class="listElement">
      <div class="listElementHeader">Second key</div>
      <button>Delete</button>
    </li>
    <li class="listElement">
      <div class="listElementHeader listElementHeaderCompromised">Compromised key</div>
      <div class="badge badge-danger">Compromised</div>
      <button>Delete</button>
    </li>
    <li class="listElement">
      <div class="listElementHeader">Last key</div>
      <button>Delete</button>
    </li>
  </ul>
</div>
`;
